patroni/api.py

```diff
@@ -110,6 +113,45 @@ def do_GET_patroni(self):
         response = self.get_postgresql_status(True)
         self._write_status_response(200, response)
 
+    def do_GET_config(self):
+        self._write_json_response(200, self.server.patroni.config.dynamic_configuration)
+
+    @staticmethod
+    def _patch_config(config, data):
+        for name, value in data.items():
+            if isinstance(value, dict) and name in config:
+                RestApiHandler._patch_config(config[name], value)
+            elif value is None:
+                config.pop(name, None)
+            else:
+                config[name] = value
+
+    @check_auth
+    def do_PATCH_config(self):
+        content_length = int(self.headers.get('content-length', 0))
+        request = json.loads(self.rfile.read(content_length).decode('utf-8'))
+        cluster = self.server.patroni.ha.dcs.get_cluster()
+        data = cluster.config.data.copy()
+        RestApiHandler._patch_config(data, request)
+        if data != cluster.config.data:
```

Comment:
"stripping the quotes when necessary" -- and how are you going to compare `int` and `string`?
Unfortunately there is no simple solution for this task, because one need to know data type for each parameter or at least to know data types of some of the parameters, like "ttl": int, "loop_wait": int, "use_slots": bool and so on. Everything except could be compared after casting old and new value to string.

Reload of everything is done asynchronously, on the next iteration of HA loop.

`POST /reload` endpoint is an equivalent of SIGHUP, but again, it is not performing real `pg_ctl reload`, but re-reads config.yml. Actual reload will happen asynchronously, on the next iteration.